src/Raven.Studio/typescript/viewmodels/manage/trafficWatch.ts

```diff
@@ -13,36 +13,93 @@ import virtualColumn = require("widgets/virtualGrid/columns/virtualColumn");
+            if (item.TrafficWatchType !== "Http" || (item as TrafficWatchHttpChange).Type !== "MultiGet") {
+                runQueryFeature.executeQuery(value, item);
+                return;
+            }
 
-            const db = databasesManager.default.getDatabaseByName(item.DatabaseName);
+            const queryList = this.queryList;
             
-            recentQueriesStorage.saveAndNavigate(db, queryDto, { newWindow: true });
+            if (queryList.length === 1) {
+                runQueryFeature.executeQuery(queryList[0], item);
+            } else {
+                app.showBootstrapDialog(new trafficWatchQueriesDialog(queryList))
+                    .done(queryToExecute => {
+                        if (queryToExecute) {
+                            runQueryFeature.executeQuery(queryToExecute, item);
+                        }
+                    });
+            }
         });
     }
+
+    private static getMultiGetQueriesList(value: any): string[] {
+        const queryList: string[] = [];
+
+        const lines = value.split('\n');
+        lines.forEach((line: string) => {
+            if (line) {
+                const jsonObj = JSON.parse(line);
+                
+                const queriesEndpoint = jsonObj.Url.slice("/queries".length * -1) === "/queries";
+                const query = jsonObj.Query.slice("?query=".length);
+                                
+                if (queriesEndpoint && query) {
+                    queryList.push(query);
+                }
+            }
+        });
+
+        return queryList;
+    }
+
+    private static executeQuery(value: any, item: Raven.Client.Documents.Changes.TrafficWatchChangeBase): void {
+        const query = queryCriteria.empty();
+
+        query.queryText(value);
+        query.name("Traffic watch query");
+        query.recentQuery(true);
+
+        const queryDto = query.toStorageDto();
+
+        const db = databasesManager.default.getDatabaseByName(item.DatabaseName);
+
+        recentQueriesStorage.saveAndNavigate(db, queryDto, { newWindow: true });
+    }
     
-    syntax(column: virtualColumn, escapedValue: any) {
-        if (column.header === "Custom Info" && escapedValue !== generalUtils.escapeHtml("N/A")) {
-            return `<button class="btn btn-default btn-sm run-query"><i class="icon-query"></i><span>Run Query</span></button>`;    
-        } else {
+    syntax(column: virtualColumn, escapedValue: any, element: any): string {
+        if (column.header !== "Custom Info" || escapedValue === generalUtils.escapeHtml("N/A")) {
             return "";
         }
+        
+        this.queryList = [];
+        let buttonText: string;
+
+        if (element.Type === "MultiGet") {
+            this.queryList = runQueryFeature.getMultiGetQueriesList(generalUtils.unescapeHtml(escapedValue));
+            buttonText = this.queryList.length > 1 ? "RunQuery..." : (this.queryList.length === 1 ? "RunQuery" : "");
```

Comment:
do we need space between run and query on button?

Reply:
yes... fixed